app/controllers/concerns/decidim/kids/needs_tutor_authorization.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Kids
+    module NeedsTutorAuthorization
+      extend ActiveSupport::Concern
+
+      included do
+        before_action do
+          if tutor_adapter.blank?
+            flash[:alert] = t("user_minors.no_tutor_authorization", scope: "decidim.kids")
+            redirect_to decidim.account_path
+          end
+        end
+
+        before_action except: [:unverified] do
+          enforce_permission_to :index, :minor_accounts
+          redirect_to unverified_user_minors_path unless tutor_verified?
+        end
+
+        helper_method :minors, :tutor_adapter
+
+        def permission_class_chain
+          [::Decidim::Kids::Permissions] + super
+        end
+
+        private
+
+        def tutor_verified?
+          @tutor_verified ||= begin
+            authorization = granted_authorizations(current_user).where(name: current_organization.tutors_authorization)
+            authorization.any? && !authorization.first.expired?
```

Comment:
We could use the `minor_authorization` method to perform a "find_by" instead of using "where + first"

```suggestion
            authorization = minor_authorization(current_user)
            authorization && !authorization.expired?
```

Reply:
unfortunately this doesn't work because we need to check the tutor's authorization here, not the minor's. But I've rearrange the code.